api/internal/crawl/cmd/crawler/crawler.go

```diff
@@ -23,11 +24,31 @@ const (
 	retryCount           = 3
 	githubUserEnv        = "GITHUB_USER"
 	githubRepoEnv        = "GITHUB_REPO"
+	crawlIndexOnlyEnv    = "CRAWL_INDEX_ONLY"
+	crawlGithubOnlyEnv   = "CRAWL_GITHUB_ONLY"
 )
 
+// countEnvs count the environment variables whose values are not empty.
+func countEnvs(envs ...string) int {
+	count := 0
+	for _, env := range envs {
+		if env != "" {
+			count++
+		}
+	}
+	return count
+}
+
 func main() {
 	githubUser := os.Getenv(githubUserEnv)
 	githubRepo := os.Getenv(githubRepoEnv)
+	crawlIndexOnly := os.Getenv(crawlIndexOnlyEnv)
+	crawlGithubOnly := os.Getenv(crawlGithubOnlyEnv)
```

Comment:
the "mode" we are in is confusing me.


maybe define it as a thing, and take it off the command line, e.g.

```
type CrawlMode int
const (
  CrawlUnknown CrawlMode = iota
  CrawlUser
  CrawlRepo
  CrawlIndex
  CrawlGithub
)
func NewCrawlMode(s string) CrawlMode {
 switch s {
  case "user":  return CrawlUser
  etc.
 default: return CrawlUnknown
}

func main() {
  mode := NewCrawlMode(os.Args[1])
  if mode == CrawlUnknown {
    log.Fatal("must specify known mode"
 }
```

then depending on the mode we are in, demand other arguments or env vars, e.g.

if mode == CrawlUser && os.Getenv(GithubUserEnv) == "" {
  log.Fatal...
}

Reply:
Fixed it!